lib/core/src/lib/dialogs/unsaved-changes-dialog/unsaved-changes.guard.spec.ts

```diff
@@ -17,20 +17,52 @@
 
 import { TestBed } from '@angular/core/testing';
 import { MatDialog, MatDialogModule, MatDialogRef } from '@angular/material/dialog';
-import { Observable, Subject } from 'rxjs';
+import { EMPTY, Observable, of, Subject } from 'rxjs';
 import { UnsavedChangesDialogComponent } from './unsaved-changes-dialog.component';
 import { UnsavedChangesGuard } from './unsaved-changes.guard';
+import { AppConfigService, AuthenticationService, AuthGuardService, AuthService, RedirectAuthService } from '@alfresco/adf-core';
+import { HttpClientTestingModule } from '@angular/common/http/testing';
+import { AUTH_CONFIG, DateTimeProvider, OAuthLogger, OAuthService, OAuthStorage, UrlHelperService } from 'angular-oauth2-oidc';
+import { AUTH_MODULE_CONFIG } from '../../auth/oidc/auth-config';
 
 describe('UnsavedChangesGuard', () => {
     let guard: UnsavedChangesGuard;
+    let authService: AuthenticationService;
+    let appConfigService: AppConfigService;
 
     beforeEach(() => {
         TestBed.configureTestingModule({
-            imports: [MatDialogModule]
+            imports: [MatDialogModule, HttpClientTestingModule],
+            providers: [
+                UrlHelperService,
```

Comment:
I think you don't need these all providers so you can remove them all. Also you don't need HttpClientTestingModule. Instead you can just import NoopAuthModule in imports array and it should work as well as it's special module for mocked auth things